homeassistant/components/statistics/sensor.py

```diff
@@ -480,3 +455,162 @@ async def _initialize_from_database(self):
+        return STATE_UNKNOWN
+
+    def _stat_average_step(self):
+        if len(self.states) >= 2:
+            area = 0
+            for i in range(1, len(self.states)):
+                area += (
+                    self.states[i - 1]
+                    * (self.ages[i] - self.ages[i - 1]).total_seconds()
+                )
+            age_range_seconds = (self.ages[-1] - self.ages[0]).total_seconds()
+            return area / age_range_seconds
+        return STATE_UNKNOWN
+
+    def _stat_average_timeless(self):
+        return self._stat_mean()
+
+    def _stat_change(self):
+        if len(self.states) > 0:
+            return self.states[-1] - self.states[0]
+        return STATE_UNKNOWN
+
+    def _stat_change_sample(self):
+        if len(self.states) > 1:
+            return (self.states[-1] - self.states[0]) / (len(self.states) - 1)
+        return STATE_UNKNOWN
+
+    def _stat_change_second(self):
+        if len(self.states) > 1:
+            age_range_seconds = (self.ages[-1] - self.ages[0]).total_seconds()
+            if age_range_seconds > 0:
+                return (self.states[-1] - self.states[0]) / age_range_seconds
+        return STATE_UNKNOWN
+
+    def _stat_count(self):
+        return len(self.states)
+
+    def _stat_datetime_newest(self):
+        if len(self.states) > 0:
+            return self.ages[-1]
+        return STATE_UNKNOWN
+
+    def _stat_datetime_oldest(self):
+        if len(self.states) > 0:
+            return self.ages[0]
+        return STATE_UNKNOWN
+
+    def _stat_distance_95p(self):
+        if len(self.states) >= 2:
+            return 2 * 1.96 * self._stat_standard_deviation()
+        return STATE_UNKNOWN
+
+    def _stat_distance_99p(self):
+        if len(self.states) >= 2:
+            return 2 * 2.58 * self._stat_standard_deviation()
+        return STATE_UNKNOWN
+
+    def _stat_distance_absolute(self):
+        if len(self.states) > 0:
+            return max(self.states) - min(self.states)
+        return STATE_UNKNOWN
+
+    def _stat_mean(self):
+        if len(self.states) > 0:
+            return statistics.mean(self.states)
+        return STATE_UNKNOWN
+
+    def _stat_median(self):
+        if len(self.states) > 0:
+            return statistics.median(self.states)
+        return STATE_UNKNOWN
+
+    def _stat_noisiness(self):
+        if len(self.states) >= 2:
+            diff_sum = sum(
+                abs(j - i) for i, j in zip(list(self.states), list(self.states)[1:])
+            )
+            return diff_sum / (len(self.states) - 1)
+        return STATE_UNKNOWN
+
+    def _stat_quantiles(self):
+        if len(self.states) > self._quantile_intervals:
+            return [
+                round(quantile, self._precision)
+                for quantile in statistics.quantiles(
+                    self.states,
+                    n=self._quantile_intervals,
+                    method=self._quantile_method,
+                )
+            ]
```

Comment:
We don't allow tuples or lists in the state, so the list will be converted to a string.
How can the user use the data in a reasonable way?

Reply:
Totally agree but I did not contribute that PR and think it's shit. https://github.com/home-assistant/core/pull/52189
So far I didn't touch it. Might improve it in a future PR

Reply:
OK, let's leave it like this for now then